stdlib/public/stubs/CommandLine.cpp

```diff
@@ -122,24 +122,30 @@ char ** _swift_stdlib_getUnsafeArgvArgc(int *outArgLen) {
                                            szArgList[i], -1, nullptr, 0,
                                            nullptr, nullptr);
     if (szBufferSize == 0) {
-      for (char *arg : argv)
+      for (char *arg : argv) {
         free(arg);
-      return nullptr;
+      }
+
+      break;
     }
 
     char *buffer = static_cast<char *>(calloc(static_cast<size_t>(szBufferSize),
                                               sizeof(char)));
     if (buffer == nullptr) {
-      for (char *arg : argv)
+      for (char *arg : argv) {
         free(arg);
-      return nullptr;
+      }
+
+      break;
     }
 
     if (!WideCharToMultiByte(CP_UTF8, WC_ERR_INVALID_CHARS, szArgList[i], -1,
                              buffer, szBufferSize, nullptr, nullptr)) {
-      for (char *arg : argv)
+      for (char *arg : argv) {
         free(arg);
-      return nullptr;
+      }
+
+      break;
```

Comment:
@compnerd I wasn't really sure what to do specifically. If this behavior is not acceptable, we need to clear the argument vector and break instead.

Reply:
A failure here indicates that there is a Unicode conversion error.  I think that the better approach here is to match the behaviour with Darwin if you were given invalid UTF-8 encoded data on the command line somehow.  Logging the result of `GetLastError` is probably also useful in that scenario.  I think its more about understanding what Darwin does in the case that you have an invalid input somehow as both of these points that you are changing are failure paths which aren't well supported today.

Reply:
It's pretty hard to convince Darwin to pass you poorly-encoded data, but it could happen with the right mix of locale settings and a terminal that can handle that (Terminal.app is not one of those applications, it seems). In such a case, we're subject to whatever the kernel decides to hand us, which we can hope are at least valid C strings. `String(cString:)` will produce encoding artifacts for such strings, but at least we'd parse *something*.

I agree we could log `GetLastError` - does that mean we should continue parsing after here?

Reply:
We could also defer the responsibility for the conversion to the standard library.